modules/@apostrophecms/express/index.js

```diff
@@ -215,6 +203,78 @@ module.exports = {
       },
       sessions: expressSession(self.getSessionOptions()),
       cookieParser: cookieParser(),
+      apiKeys(req, res, next) {
+        const key = req.query.apikey || req.query.apiKey || getAuthorizationApiKey();
+        let taskReq;
+        if (!key) {
+          return next();
+        }
+        if (_.has(self.options.apiKeys, key)) {
+          const info = self.options.apiKeys[key];
+          if (info.role === 'admin') {
+            taskReq = self.apos.task.getReq();
+          } else {
+            taskReq = self.apos.task.getAnonReq();
+          }
+          req.user = taskReq.user;
+          req.csrfExempt = true;
+          return next();
+        }
+        return res.status(403).send({ error: 'invalid api key' });
```

Comment:
Not mandatory, but it would ease reading the code to drop an error `if (key && !_.has(self.options.apiKeys, key))` otherwise `return next()`

Reply:
I'll put in a check that self.options.apiKeys is actually set so you can't crash it with this, but otherwise keep it the same